pkg/client/outofband/client.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package outofband
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/protocol/decorator"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+)
+
+const (
+	protocolURI    = "https://didcomm.org/oob-request/1.0"
+	requestMsgType = protocolURI + "/request"
+)
+
+// RequestOptions allow you to customize the way request messages are built.
+type RequestOptions func(*Request) error
+
+// ConnectionRecorder records connection records produced as byproducts of creation of messages.
+type ConnectionRecorder interface {
+	SaveInvitation(id string, i interface{}) error
+}
+
+// Provider provides the dependencies for the client.
+type Provider interface {
+	// DidDocServiceFunc returns a function that returns a DID doc `service` entry.
+	// Used when no service entries are specified when creating messages.
+	DidDocServiceFunc() func() (*did.Service, error)
+	ConnRecorder() ConnectionRecorder
+}
+
+// Client for the Out-Of-Band protocol:
+// https://github.com/hyperledger/aries-rfcs/blob/master/features/0434-outofband/README.md
+type Client struct {
+	didDocSvcFunc func() (*did.Service, error)
+	connRecorder  ConnectionRecorder
+}
+
+// New returns a new Client for the Out-Of-Band protocol.
+func New(p Provider) *Client {
+	return &Client{
+		didDocSvcFunc: p.DidDocServiceFunc(),
+		connRecorder:  p.ConnRecorder(),
```

Comment:
Why can't you use, `connection.NewRecorder(ctx)` here? And use provider matching aries context.

Reply:
the context currently doesn't provide a connection recorder. Maybe can be solved in #1467

Reply:
It also doesn't provide a function that creates a did doc `service` object (requires new keys and router info (if applicable)). This capability is also needed in the didexchange protocol service.